### PR TITLE
Fix docs: user storage spi jpa quickstart description

### LINKED ISSUE #30941: Fix docs about User Storage SPI JPA quickstart
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

docs

### Describe the bug

In table "Table 1. User Storage SPI Quickstarts" of chapter "User Storage SPI" [1] the documentation says "Demonstrates implementing a user storage provider using EJB and JPA"

But we no longer have EJB. Then it must be adjusted to "Demonstrates implementing a user storage provider using JPA"

[1] https://www.keycloak.org/docs/latest/server_development/index.html#_user-storage-spi 

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

Docs should say: "Demonstrates implementing a user storage provider using JPA"

### Actual behavior

Docs says: "Demonstrates implementing a user storage provider using EJB and JPA"

### How to Reproduce?

N/A

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by gilvansfilho**
Closes #30941

### CODE REVIEW

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for this fix, and I agree it is correct. 

As of our contributing guidelines we require a signoff in every commit, see `CONTRIBUTING.md`. Once this is there, this PR is ready for merging.

You can update the commit message of this PR to include the 

```
Signed-off-by: Author Name <authoremail@example.com>
```

**Comment by gilvansfilho** [comment]
> Thank you for this fix, and I agree it is correct.
> 
> As of our contributing guidelines we require a signoff in every commit, see `CONTRIBUTING.md`. Once this is there, this PR is ready for merging.
> 
> You can update the commit message of this PR to include the
> 
> ```
> Signed-off-by: Author Name <authoremail@example.com>
> ```

@ahus1  Done :)

**Review by ahus1 — APPROVED** [review]
Thank you for this fix!

**Pull request merged**
